stdlib/public/core/Slice.swift

```diff
@@ -229,6 +229,23 @@ extension Slice: Collection {
   }
 }
 
+extension Slice {
+  @inlinable
+  public __consuming func _copyContents(
+      initializing buffer: UnsafeMutableBufferPointer<Element>
+  ) -> (Iterator, UnsafeMutableBufferPointer<Element>.Index) {
+    if let (_, copied) = self.withContiguousStorageIfAvailable({
+      $0._copyContents(initializing: buffer)
+    }) {
+      let distance = buffer.distance(from: buffer.startIndex, to: copied)
+      let i = index(startIndex, offsetBy: distance)
```

Comment:
`distance` is just a fancier spelling for `copied` -- we shouldn't make things appear more complicated as they are. :-) 

```suggestion
      let i = index(startIndex, offsetBy: copied)
```

Reply:
I wonder about this `index(offsetBy:)` call -- will this make things slower if `Self` is only a forward collection?

Reply:
Regarding `distance` and `copied`, if I remember correctly there was a type-matching issue. I'll take another look.

Reply:
`_copyContents` declares the second component of its return type as `UnsafeBufferPointer.Index`, but that's just a typealias for `Int`!

Reply:
I can't reproduce the type issue there. Might have been a playground quirk. I'll incorporate this with a better variable name.